packages/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button, Typography, WithStyles, withStyles } from "@material-ui/core";
+import { ButtonProps } from "@material-ui/core/Button";
+import clsx from "clsx";
+import * as React from "react";
+
+import { FilterBarActiveFilterBadge, FilterBarActiveFilterBadgeProps } from "../filterBarActiveFilterBadge/FilterBarActiveFilterBadge";
+import { FilterBarButtonClassKey, styles } from "./FilterBarButton.styles";
+
+export interface FilterBarButtonProps extends ButtonProps {
+    dirtyFieldsBadge?: React.ComponentType<FilterBarActiveFilterBadgeProps>;
+    countValue?: number;
+    openPopover: boolean;
+}
+
+const FilterBarButton = ({
+    children,
+    dirtyFieldsBadge,
+    countValue,
+    openPopover,
+    classes,
+    ...buttonProps
+}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+    const showFilterBadge = !!(countValue && countValue > 0);
+    const FilterBarActiveFilterBadgeComponent = dirtyFieldsBadge ? dirtyFieldsBadge : FilterBarActiveFilterBadge;
+
+    return (
+        <Button
+            className={clsx(classes.root, showFilterBadge && classes.selected && classes.withFilterBadge, openPopover && classes.open)}
+            disableRipple
+            {...buttonProps}
+        >
+            <div className={clsx(classes.labelWrapper, showFilterBadge && classes.labelWrapperWithValues)}>
+                <Typography variant="body1">{children}</Typography>
+            </div>
```

Comment:
`<div />` and `<p />` are technically not allowed inside a `<button />`.  

Maybe it makes sense to use `<Button variant="outline" />` which we currently don't use anyway. 
Then you could probably move all of the Styling (except the badge-stuff) to the theme, and you probably don't need the `labelWrapper` and `Typography` as you can style `.MuiButton-label` directly.

Reply:
Done